sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusClientBuilder.java

```diff
@@ -570,20 +631,168 @@ public ServiceBusReceiverClientBuilder subscriptionName(String subscriptionName)
+                    connectionProcessor, retryOptions.getTryTimeout(), tracerProvider, messageSerializer,
+                    receiverOptions);
+
+                return new ServiceBusReceiverAsyncClient(connectionProcessor.getFullyQualifiedNamespace(), entityPath,
+                    entityType, receiverOptions, connectionProcessor, ServiceBusConstants.OPERATION_TIMEOUT,
+                    tracerProvider, messageSerializer, ServiceBusClientBuilder.this::onClientClose, manager);
+            } else {
+                return new ServiceBusReceiverAsyncClient(connectionProcessor.getFullyQualifiedNamespace(), entityPath,
+                    entityType, receiverOptions, connectionProcessor, ServiceBusConstants.OPERATION_TIMEOUT,
+                    tracerProvider, messageSerializer, ServiceBusClientBuilder.this::onClientClose);
+            }
+        }
+
+        /**
+         * Creates a <b>synchronous</b>, <b>session-aware</b> Service Bus receiver responsible for reading {@link
+         * ServiceBusMessage messages} from a specific queue or topic.
+         *
+         * @return An new {@link ServiceBusReceiverClient} that receives messages from a queue or topic.
+         * @throws IllegalStateException if {@link #queueName(String) queueName} or {@link #topicName(String)
+         *     topicName} are not set or, both of these fields are set. It is also thrown if the Service Bus {@link
+         *     #connectionString(String) connectionString} contains an {@code EntityPath} that does not match one set in
+         *     {@link #queueName(String) queueName} or {@link #topicName(String) topicName}. Lastly, if a {@link
+         *     #topicName(String) topicName} is set, but {@link #subscriptionName(String) subscriptionName} is not.
+         * @throws IllegalArgumentException Queue or topic name are not set via {@link #queueName(String)
+         *     queueName()} or {@link #topicName(String) topicName()}, respectively.
+         */
+        public ServiceBusReceiverClient buildClient() {
+            return new ServiceBusReceiverClient(buildAsyncClient(), retryOptions.getTryTimeout());
+        }
+
+        /**
+         * This is a rolling session receiver only if maxConcurrentSessions is > 0 AND sessionId is null or empty. If
+         * there is a sessionId, this is going to be a single, named session receiver.
+         *
+         * @return {@code true} if this is an unnamed rolling session receiver; {@code false} otherwise.
+         */
+        private boolean isRollingSessionReceiver() {
+            if (maxConcurrentSessions == null) {
+                return false;
+            }
+
+            if (maxConcurrentSessions < 1) {
+                throw logger.logExceptionAsError(
+                    new IllegalArgumentException("Maximum number of concurrent sessions must be positive."));
+            }
+
+            return CoreUtils.isNullOrEmpty(sessionId);
```

Comment:
The use case: where user want to receive message from one and only one unnamed session and No rollover, this should  return false here if  `CoreUtils.isNullOrEmpty(sessionId)`

Reply:
In that case, and from Srikanta's gist, they won't get rollover because they didn't set maxConcurrentSessions. That check is done a few lines up.

Reply:
https://gist.github.com/hemanttanwar/f7facad764b087dcf05eb4350ce85ea6